packages/dataviews/src/components/dataviews-context/index.ts

```diff
@@ -11,18 +11,18 @@ import type { SetSelection } from '../../private-types';
 import { LAYOUT_TABLE } from '../../constants';
 
 type DataViewsContextType< Item > = {
-	view: View;
-	onChangeView: ( view: View ) => void;
+	view?: View;
+	onChangeView?: ( view: View ) => void;
```

Comment:
For me view should be mandatory, onChangeView could potentially be optional but that requires deep changes to the DataViews layouts to disable sorting, fitlering... so for now, I think we should keep all the existing props untouched. (including paginationInfo, onChangeSelection)

Not rendering the DataViews layout (table, or list... ) doesn't mean these props should be made optional, it just means that the layout will be rendered manually at the right position. The layouts still expect these things to be defined.

Reply:
can we make it required by the Context, but optional for the component passing a default `view = { type: LAYOUT_TABLE }` prop? https://github.com/Automattic/wp-calypso/pull/102996/commits/b785dd4320ff0fb39d44acdcb111f8d72467b0c7

It ensures a defined `view` in the context, but is optional for the consumer when creating DataViews instances

Reply:
It's similar to the `fields[]` prop, [required by the context](https://github.com/Automattic/wp-calypso/blob/c5ec43891c6abec612d9fa1504a9dadb9d1a0b10/packages/dataviews/src/components/dataviews-context/index.ts#L16), but optional by the component with a default empty array value.

Reply:
For me a `DataViews` component without fields or without view doesn't really make sense. So I'd return the question to you, why do you want a DataViews component without view or fields, what would you use that for?